keel-retrofit/src/main/kotlin/com/netflix/spinnaker/keel/retrofit/KeelRetrofitConfiguration.kt

```diff
@@ -17,73 +17,21 @@ package com.netflix.spinnaker.keel.retrofit
 
 import com.netflix.spinnaker.config.OkHttp3ClientConfiguration
 import com.netflix.spinnaker.fiat.shared.FiatPermissionEvaluator
-import com.netflix.spinnaker.okhttp.OkHttp3MetricsInterceptor
 import com.netflix.spinnaker.okhttp.OkHttpClientConfigurationProperties
-import java.util.concurrent.TimeUnit.MILLISECONDS
-import okhttp3.ConnectionPool
-import okhttp3.Interceptor
-import okhttp3.OkHttpClient
 import okhttp3.logging.HttpLoggingInterceptor
-import org.slf4j.LoggerFactory
 import org.springframework.beans.factory.annotation.Value
 import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean
 import org.springframework.boot.context.properties.EnableConfigurationProperties
 import org.springframework.context.annotation.Bean
 import org.springframework.context.annotation.Configuration
 import org.springframework.context.annotation.Import
-import org.springframework.context.annotation.Primary
+import org.springframework.context.annotation.Lazy
 
 @Configuration
 @Import(OkHttp3ClientConfiguration::class)
 @EnableConfigurationProperties(OkHttpClientConfigurationProperties::class, KeelRetrofitProperties::class)
 class KeelRetrofitConfiguration {
 
-  private val log = LoggerFactory.getLogger(javaClass)
-
-  @Bean(name = ["retrofitClient", "okClient"])
-  @Primary // TODO: (SR) -- Refactor it to use the new client provider.
-  fun retrofitClient(
-    okHttpClientConfig: OkHttp3ClientConfiguration,
-    okHttpClientProperties: OkHttpClientConfigurationProperties,
-    retrofitProperties: KeelRetrofitProperties,
-    interceptors: Set<Interceptor>?
-  ): OkHttpClient {
-    val builder = okHttpClientConfig
-      .create()
-      .apply {
-        networkInterceptors()
-          .add(Interceptor { chain ->
-            chain.proceed(
-              chain.request().newBuilder()
-                .header("User-Agent", retrofitProperties.userAgent)
-                .build()
-            )
-          })
-        interceptors?.forEach {
-          log.info("Adding OkHttp Interceptor: ${it.javaClass.simpleName}")
-          addInterceptor(it)
-        }
-        connectionPool(ConnectionPool(
-          okHttpClientProperties.connectionPool.maxIdleConnections,
-          okHttpClientProperties.connectionPool.keepAliveDurationMs.toLong(),
-          MILLISECONDS
-        ))
-        retryOnConnectionFailure(okHttpClientProperties.isRetryOnConnectionFailure)
-      }
-
-    builder.interceptors().also {
-      // If the metrics interceptor (which complains about X-SPINNAKER-* auth headers missing in the request)
```

Comment:
Is this ordering problem not an issue with the new client provider?

Reply:
I made the SpinnakerHeadersInterceptor to go first in the chain so that OkHttp3Metrics interceptor can take advantage of the data we are adding.